python/src/uagents/communication.py

```diff
@@ -163,10 +163,12 @@ async def send_exchange_envelope(
                     success = resp.status == 200
                     if success:
                         if sync:
-                            # If the message is synchronous but not verified, return the envelope
                             env = Envelope.model_validate(await resp.json())
-                            if env.signature is None:
-                                return env
```

Comment:
This return is redundant because it should only apply when the message is sent from somethere other than an agent (i.e. the `query` function). In this case, the dispatcher should have no sinks and the function `dispatch_sync_response_envelope` will return the envelope.